frontend/src/diagram/DiagramWebSocketContainer.tsx

```diff
@@ -617,6 +631,52 @@ export const DiagramWebSocketContainer = ({
     }
   };
 
+  const handleError = useCallback(
+    (loading, data, error) => {
+      let errorMessage = undefined;
+      if (!loading) {
+        if (error) {
+          errorMessage = error.message;
+        } else if (data && Object.keys.length > 0) {
```

Comment:
- Do not consider that error and data are mutually exclusive use cases, you can have both
- `Object.keys.length  > 0` huh? Why are you testing that the length of a function is not zero? It makes no sense.

Reply:
That's a mistake, it should be Object.keys(data).length > 0

Reply:
Fixed by a6408c7bd8b82b72a5055ae784e4ed553ab44120